__tests__/api/txMining.test.ts

```diff
@@ -0,0 +1,77 @@
+import MockAdapter from 'axios-mock-adapter';
+import axios from 'axios';
+import txMiningApi from '../../src/api/txMining';
+
+describe('txMiningApi', () => {
+  let mock;
+
+  beforeEach(() => {
+    mock = new MockAdapter(axios);
+  });
+
+  afterEach(() => {
+    mock.restore();
+  });
+
+  it('getJobStatus should send the right request', async () => {
+    const data = { status: 'done' };
+    mock.onGet('job-status').reply(200, data);
+
+    const result = await new Promise((resolve, reject) => {
+      txMiningApi.getJobStatus('jobId', resolve);
+    });
+
+    expect(result).toEqual(data);
+
+    expect(mock.history.get.length).toBe(1);
+    expect(mock.history.get[0].params).toEqual({ "job-id": 'jobId' });
+  });
+
+  it('cancelJob should send the right request', async () => {
+    const data = { status: 'cancelled' };
+    mock.onPost('cancel-job').reply(200, data);
+
+    const result = await new Promise((resolve, reject) => {
+      txMiningApi.cancelJob('jobId', resolve);
+    });
+
+    expect(result).toEqual(data);
+
+    expect(mock.history.post.length).toBe(1);
+    expect(mock.history.post[0].data).toEqual(JSON.stringify({ "job-id": 'jobId' }));
+  });
+
+  it('getHealth should send the right request', async () => {
+    const data = { status: 'pass' };
+    mock.onGet('health').reply(200, data);
+
+    const result = await txMiningApi.getHealth();
+
+    expect(result).toEqual(data);
+  });
+
+  it('getHealth should allow capturing errors in case of network error', async () => {
+    mock.onGet('health').networkError();
+
+    await expect(txMiningApi.getHealth()).rejects.toThrow();
```

Comment:
I like this test validating the error handling of `getHealth()`.

Should we have similar tests for `getJobStatus()`, `cancelJob()` and `submitJob()`?

Reply:
I've added some more tests in https://github.com/HathorNetwork/hathor-wallet-lib/pull/579/commits/ea887275286c208a12e434864712a8b2b2243a1a